### PR TITLE
feat: add consumption of carnet tickets

### PR DESCRIPTION
**Pull request by rosvik**
https://github.com/AtB-AS/mittatb-app/assets/1774972/a99ccbc3-49a6-462d-9434-065eb68ae85e

closes https://github.com/AtB-AS/kundevendt/issues/12562

### CODE REVIEW

File: src/ticketing/utils.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import {getLastUsedAccess} from '@atb/fare-contracts/utils';
 export function isCarnetTravelRight(
   travelRight: TravelRight | undefined,
 ): travelRight is CarnetTravelRight {
-  return travelRight?.type === 'CarnetTicket';
+  return !!travelRight && 'maximumNumberOfAccesses' in travelRight;
```
Comment (rosvik) [inline src/ticketing/utils.ts:21]: @gorandalum You've talked a lot about not using the `type` field for logic in the app: Do you think this is this an ok replacement?

File: src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+export const ConsumeCarnetBottomSheet = ({fareContractId}: Props) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<boolean>(false);
+  const {close} = useBottomSheet();
+
+  const onConsume = async () => {
```
Comment (reidzeibel) [inline src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx:29]: IIRC, we want to start using react-query for new API operations, so we should do that also for this one.

Comment (rosvik) [reply src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx:29]: I think this is the one type of requests that shouldn't use react-query. There is no state to keep up to date here, and no dependencies to keep track of. Just an user interaction that causes an API request, which doesn't need to be more complicated that this.

Comment (reidzeibel) [reply src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx:29]: Fair enough! 👍

File: src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    setIsLoading(true);
+    try {
+      await consumeCarnet(fareContractId);
+      close();
+    } catch (e: any) {
+      const errorData = getAxiosErrorMetadata(e);
+      console.error('Error when consuming carnet: ', errorData);
```
Comment (rosvik) [inline src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx:36]: Noticing we should probably handle errors better than this. I guess a bugsnag report would make sense here.

**Review by reidzeibel — APPROVED** [review]
LGTM!

File: src/fare-contracts/components/ConsumeCarnetButton.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import {TicketValid} from '@atb/assets/svg/mono-icons/ticketing';
```
Comment (marius-at-atb) [inline src/fare-contracts/components/ConsumeCarnetButton.tsx:1]: Should this file be called `ConsumeCarnetSectionItem.tsx`?

File: src/fare-contracts/components/ConsumeCarnetButton.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import {useBottomSheet} from '@atb/components/bottom-sheet';
+import {LinkSectionItem, SectionItemProps} from '@atb/components/sections';
+import {ThemeIcon} from '@atb/components/theme-icon';
+import {useTheme} from '@atb/theme';
+import {FareContractTexts, useTranslation} from '@atb/translations';
+import React from 'react';
+import {ConsumeCarnetBottomSheet} from './ConsumeCarnetBottomSheet';
+
+type ConsumeCarnetButtonProps = SectionItemProps<{
```
Comment (marius-at-atb) [inline src/fare-contracts/components/ConsumeCarnetButton.tsx:10]: Should this be called `ConsumeCarnetSectionItemProps`?

File: src/fare-contracts/components/ConsumeCarnetButton.tsx (hunk 4 above)
Comment (rosvik) [reply src/fare-contracts/components/ConsumeCarnetButton.tsx:1]: Yes, good catch! Fixed this and your other comment in 9665b1ce6fbdeba4e4aae8ba056467e3241ca58f

File: src/fare-contracts/components/ConsumeCarnetButton.tsx (hunk 5 above)
Comment (rosvik) [reply src/fare-contracts/components/ConsumeCarnetButton.tsx:10]: 9665b1ce6fbdeba4e4aae8ba056467e3241ca58f

File: src/ticketing/utils.ts (hunk 1 above)
Comment (gorandalum) [reply src/ticketing/utils.ts:21]: Yes, that's great, directly from the source :+1:

File: src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx (hunk 2 above)
Comment (gorandalum) [reply src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx:30]: We should use react-query. Then we get error/loading statues, retry possibility, and more out of the box, and we get the same typing/behaviour on all network requests.

File: src/fare-contracts/components/ConsumeCarnetButton.tsx (hunk 5 above)
Comment (gorandalum) [reply src/fare-contracts/components/ConsumeCarnetButton.tsx:10]: Also note that I think just using `Props` is fine when it is not exported.

**Pull request merged**